packages/ses/src/module-link.js

```diff
@@ -0,0 +1,62 @@
+import { makeModuleInstance } from './module-instance.js';
+
+const { entries } = Object;
+// q, as in quote, for quoting strings in error messages.
+const q = JSON.stringify;
+
+// `link` creates `ModuleInstances` and `ModuleNamespaces` for a module and its
+// transitive dependencies and connects their imports and exports.
+// After linking, the resulting working set is ready to be executed.
+// The linker only concerns itself with module namespaces that are objects with
+// property descriptors for their exports, which the Compartment proxies with
+// the actual `ModuleNamespace`.
+export const link = (
+  compartment,
+  compartmentPrivateFields,
+  moduleSpecifier,
+) => {
+  const { moduleRecords, instances } = compartmentPrivateFields.get(
+    compartment,
+  );
+
+  // Memoize.
+  if (instances.has(moduleSpecifier)) {
+    return instances.get(moduleSpecifier);
+  }
+
+  const moduleRecord = moduleRecords.get(moduleSpecifier);
+  if (moduleRecord == null) {
+    throw new ReferenceError(`Missing link to module ${q(moduleSpecifier)}`);
+  }
+
+  // The module compartment record graph can refer through an alias to a record
```

Comment:
Comment is confusing, because it mixes levels of abstractions. From the wording, I expected aliases to appear here. They don't. Rather, aliases are processed in `module-load.js` to create a cross-compartment linked graph of module compartment records.

Reply:
I posted a follow-on commit that changes this module to use mutually recursive `link` and `instantiate` functions, so the module specifier and compartment in scope are unambiguous. The new comment simply says that the call to `instantiate` may transit to a different compartment. `link` occurs with the dependee’s compartment and specifier in scope; `instantiate` occurs with the dependency’s compartment and specifier in scope.